src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java

```diff
@@ -388,22 +366,48 @@ private void startTestItem(ExtensionContext context, List<Object> arguments, Ite
 		}).collect(Collectors.toList());
 	}
 
+	private Maybe<String> startBeforeAfter(Method method, ExtensionContext parentContext, ExtensionContext context, ItemType itemType) {
+		Launch launch = getLaunch(context);
+		StartTestItemRQ rq = new StartTestItemRQ();
+		rq.setStartTime(Calendar.getInstance().getTime());
+		rq.setName(method.getName() + "()");
+		rq.setDescription(method.getName());
+		String uniqueId = parentContext.getUniqueId() + "/[method:" + method.getName() + "()]";
+		rq.setUniqueId(uniqueId);
+		ofNullable(context.getTags()).ifPresent(it -> rq.setAttributes(it.stream()
+				.map(tag -> new ItemAttributesRQ(null, tag))
+				.collect(Collectors.toSet())));
+		rq.setType(itemType.name());
+		rq.setRetry(false);
+		String codeRef = method.getDeclaringClass().getCanonicalName() + "." + method.getName();
+		rq.setCodeRef(codeRef);
+		TestCaseIdEntry testCaseIdEntry = ofNullable(method.getAnnotation(TestCaseId.class)).map(TestCaseId::value)
+				.map(TestCaseIdEntry::new)
+				.orElseGet(() -> getTestCaseId(codeRef));
+		rq.setTestCaseId(testCaseIdEntry.getId());
+		Maybe<String> itemId = launch.startTestItem(idMapping.get(parentContext), rq);
+		StepAspect.setParentId(itemId);
+		return itemId;
+	}
+
 	private @NotNull TestCaseIdEntry getTestCaseId(@NotNull String codeRef) {
-		return new TestCaseIdEntry(codeRef, codeRef.hashCode());
+		return new TestCaseIdEntry(codeRef);
 	}
 
 	private @NotNull TestCaseIdEntry getTestCaseId(@NotNull Method method, String codeRef, List<Object> arguments) {
 		TestCaseId caseId = method.getAnnotation(TestCaseId.class);
 		if (caseId != null) {
 			return caseId.parametrized() ?
 					TestCaseIdUtils.getParameterizedTestCaseId(method, arguments) :
-					new TestCaseIdEntry(caseId.value(), caseId.value().hashCode());
+					new TestCaseIdEntry(caseId.value());
 		}
 		return getTestCaseId(codeRef, arguments);
 	}
 
 	private @NotNull TestCaseIdEntry getTestCaseId(@NotNull String codeRef, @NotNull List<Object> arguments) {
-		return new TestCaseIdEntry(StringUtils.join(codeRef, arguments), Arrays.deepHashCode(new Object[] { codeRef, arguments }));
+		boolean isParametersPresent = Objects.nonNull(arguments) && !arguments.isEmpty();
```

Comment:
arguments already not null, so `Objects.nonNull(arguments)` is redundant here.

Reply:
fixed